pkg/policy/repository.go

```diff
@@ -522,6 +522,10 @@ func (p *Repository) getMatchingRules(securityIdentity *identity.Identity) (ingr
 	ingressMatch = false
 	egressMatch = false
 	for _, r := range p.rules {
+		isNode := securityIdentity.ID == identity.ReservedIdentityHost
+		if (r.NodeSelector.LabelSelector != nil) != isNode {
```

Comment:
nit, might be slightly easier to follow like this?
```suggestion
		isNode := securityIdentity.ID == identity.ReservedIdentityHost
		selectsNode := r.NodeSelector.NodeSelector != nil
		if selectsNode != isNode {
```